src/plugins/intel_gna/src/pre_post_process/preprocessing.cpp

```diff
@@ -32,17 +32,324 @@ int8_t ConvertFloatToInt8(float src) {
 
 void ConvertToInt16(int16_t* ptr_dst,
                     const float* ptr_src,
-                    const uint32_t num_rows,
-                    const uint32_t num_columns,
+                    const size_t num_rows,
+                    const size_t num_columns,
                     const float scale_factor) {
     if (!ptr_dst || !ptr_src) {
         return;
     }
-    for (uint32_t i = 0; i < num_rows * num_columns; i++) {
+    for (size_t i = 0; i < num_rows * num_columns; i++) {
         ptr_dst[i] = ConvertFloatToInt16(ptr_src[i] * scale_factor);
     }
 }
 
+using InferenceEngine::Precision;
+
+void export_scores(void* ptr_dst,
+                   const void* ptr_src,
+                   intel_dnn_orientation_t orientation,
+                   size_t num_frames,
+                   size_t num_group,
+                   size_t num_vector_elements,
+                   size_t num_active_elements,
+                   size_t num_vector_stride,
+                   const InferenceEngine::Precision& precision_in,
+                   const InferenceEngine::Precision& precision_out,
+                   const float scale_factor,
+                   bool isAvx2Supported) {
+    if (ptr_src == nullptr || ptr_dst == nullptr) {
+        THROW_GNA_EXCEPTION << "Received null pointer arguments";
+    }
+    if (precision_out != Precision::I32 && precision_out != Precision::FP32) {
+        THROW_GNA_EXCEPTION << "Unsupported target precision for infer : " << precision_out.name();
+    }
+
+    bool transpose = orientation == kDnnInterleavedOrientation && num_group > 1 && num_vector_stride > 1;
+    // TODO: AVX2 support for zero-padding isn't implemented yet;
+    // fall back to the non-vectorized version when it's necessary
+    bool needsZeroPadding = (num_active_elements != num_vector_stride) || (num_frames != num_group);
+
+    switch (precision_out) {
+    case Precision::FP32:
+        switch (precision_in) {
+        case Precision::I8:
+            if (isAvx2Supported && !needsZeroPadding) {
+                convert_matrix_int8_to_fp32_avx(reinterpret_cast<float*>(ptr_dst),
+                                                reinterpret_cast<const int8_t*>(ptr_src),
+                                                num_vector_stride,
+                                                num_frames,
+                                                scale_factor,
+                                                transpose);
+                break;
```

Comment:
I suggest using `if() {} else {}` here and one `break` at the end. It took me a while to understand that `convert_matrix_int8_to_fp32_avx` and `unscale_transpose_and_cast` are mutually exclusive. 

The same applies to the rest cases in the function.